pkg/git/v2/executor.go

```diff
@@ -31,6 +33,20 @@ type executor interface {
 // Censor censors content to remove secrets
 type Censor func(content []byte) []byte
 
+// censorURLCredentials censors credentials in URLs
+// It replaces the password/token part of URLs with "xxxxx"
+func censorURLCredentials(s string) string {
+	if u, err := url.Parse(s); err == nil && u.User != nil {
+		return u.Redacted()
+	}
+
+	// Fallback to regex for URLs embedded in command arguments
+	// Match URLs like git clone https://username:token@gitlab.com:443/group/project.git
+	// Groups: 1:(https://username:) 2:(token) 3:(@gitlab.com:443)
+	re := regexp.MustCompile(`(https?://[^:]+:)([^@]+)(@[^/\s:]+(?::[0-9]+)?)`)
```

Comment:
Compilation should be extracted to be done just once, not in every `censorURLCredentials` call

Reply:
https://github.com/kubernetes-sigs/prow/pull/513